src/content/learn/typescript.md

```diff
@@ -0,0 +1,474 @@
+---
+title: Using TypeScript
+re: https://github.com/reactjs/react.dev/issues/5960
+---
+
+<Intro>
+
+TypeScript is a popular way to add type definitions to JavaScript codebases. Out of the box, TypeScript [supports JSX](/learn/writing-markup-with-jsx) and you can get full React Web support by adding [`@types/react`](https://www.npmjs.com/package/@types/react) and [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) to your project.
+
+</Intro>
+
+<YouWillLearn>
+
+* [TypeScript with React Components](/learn/typescript#typescript-with-react-components)
+* [Typing common hooks](/learn/typescript#typing-hooks)
+* [Common types from `@types/react`](/learn/typescript/#useful-types)
+* [Further learning locations](/learn/typescript/#further-learning)
+
+</YouWillLearn>
+
+
+All of the production-grade frameworks mentioned in [Start a New React Project](/learn/start-a-new-react-project) offer support for using TypeScript with React, we recommend consulting their documentation for more information on setup. This guide assumes you have your project configured to support writing `*.tsx` TypeScript React files, and have finished the Quick Start guide.
```

Comment:
If it's a single sentence, it might not need that, but I'm open to that - you don't want to be too hand-hold-y here because how setting up TypeScript on each platform is probably a bit different and can change over time